### PR TITLE
Exportable pad_max_tiles

### PR DESCRIPTION
**Pull request by lucylq**
#### Context
What is the purpose of this PR? Is it to
- [x] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
Make `pad_max_tiles` exportable. Add tests to CLIP for `pad_max_tiles` = True. 
I can also make these changes specifically for inference if that is preferred.

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [x] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [x] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [x] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

File: torchtune/modules/transforms/vision_utils/pad_dim_to_size.py [hunk 1]
```diff
@@ -28,16 +28,18 @@ def pad_dim_to_size(
...
+    # Set up 0 padding for the entire tensor.
+    # Padding is in order W*H*C*N, with front and back for each dim.
+    # https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html
+    padding = [0] * 2 * input.dim()
+    # Find the pad_index: convert NCHW to WHCN, and only pad the back
+    # (not both sides).
+    pad_index = (input.dim() - dim) * 2 - 1
+    padding[pad_index] = pad_size
+    # Pad dim to size.
+    return torch.nn.functional.pad(input, padding, value=fill)
```
Comment (pbontrager) [inline torchtune/modules/transforms/vision_utils/pad_dim_to_size.py:45]: nit: we usually do "import torch.nn.functional as F" here

**Review by pbontrager — APPROVED** [review]
Looks great! I should have used F.pad in the first place.
